### ISSUE TITLE
Bug(consistent-spacing-between-blocks): Shouldn't require space between start of for loop and test

### ISSUE THREAD

**Issue by TastyPi** [lo1tuma/eslint-plugin-mocha#406]
```ts
for (const case of cases) {
  it("checks ${case}", async function () {
    ...
  });
}
```

`mocha/consistent-spacing-between-blocks` complains that the `it` call should have a blank space before it. This ends up conflicting with our prettier config.

**Comment by KristjanTammekivi** [comment]
This doesn't happen if the loop is the first item in the `describe`, but will if there is another `it` before.
Here's a failing test for this

```typescript
        {
            code: `describe('foo', () => {
                it('bar', () => {});

                testCases.forEach((testCase) => {
                    it(testCase.title, () => {});
                });
            });`
        }
```
